src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2083,6 +2092,18 @@ SubquerySqlTaskList(Job *job)
 				/* get next fragment for the first relation list */
 				fragmentCombinationCell = lnext(fragmentCombinationCell);
 			}
+
+			prunedTableSize += ShardLength(shardInterval->shardId);
+		}
+
+		/*
+		 * The shards of the anchor table are not shipped between workers if it can be
```

Comment:
This comment does not give any clues on why we haven't used `AnchorRangeTableId ()`. Do you think does it make sense to mention performance issues here?
Also, we're not picking the largest table, we're picking largest total shard size?